src/ngx_http_datadog_module.cpp

```diff
@@ -162,32 +162,32 @@ static ngx_command_t datadog_commands[] = {
       "datadog_operation_name",
       "opentracing_operation_name",
       anywhere | NGX_CONF_TAKE1,
-      set_datadog_operation_name,
+      ngx_http_set_complex_value_slot,
       NGX_HTTP_LOC_CONF_OFFSET,
-      0,
+      offsetof(datadog_loc_conf_t, operation_name_script),
       nullptr),
 
     DEFINE_COMMAND_WITH_OLD_ALIAS(
       "datadog_location_operation_name",
       "opentracing_location_operation_name",
       anywhere | NGX_CONF_TAKE1,
-      set_datadog_location_operation_name,
+      ngx_http_set_complex_value_slot,
       NGX_HTTP_LOC_CONF_OFFSET,
-      0,
+      offsetof(datadog_loc_conf_t, loc_operation_name_script),
       nullptr),
 
     { ngx_string("datadog_resource_name"),
       anywhere | NGX_CONF_TAKE1,
-      set_datadog_resource_name,
+      ngx_http_set_complex_value_slot,
       NGX_HTTP_LOC_CONF_OFFSET,
-      0,
+      offsetof(datadog_loc_conf_t, resource_name_script),
       nullptr},
 
     { ngx_string("datadog_location_resource_name"),
       anywhere | NGX_CONF_TAKE1,
-      set_datadog_location_resource_name,
+      ngx_http_set_complex_value_slot,
       NGX_HTTP_LOC_CONF_OFFSET,
-      0,
+      offsetof(datadog_loc_conf_t, loc_resource_name_script),
```

Comment:
Is there an interest having two directives if at the end it is the same used function ?

Reply:
If you look at the "datadog_resource_name" and "datadog_location_resource_name", everything is similar except the directive name. Changing that might introduces a breaking change so it does not have to be in this PR if it is the case but one of them looks useless.

Reply:
`datadog_resource_name` set the `resource_name_script` variable, while `datadog_location_resource_name` set `loc_resource_name_script`. These are two _different_ variables,

Reply:
Well

Reply:
I read that part too quickly

Reply:
my bad